app/decorators/controllers/hyrax/admin/stats_controller_decorator.rb

```diff
@@ -0,0 +1,11 @@
+module Hyrax
+  Admin::StatsController.class_eval do
+    def show
+      super
+      @works_to_display ||= ::FileDownloadStat.all.select do |e|
```

Comment:
This will load all database records into memory and iterate over them trying to find records that match this condition. But if you do `::FileDownloadStat.where(user_id: current_user.id)` then the database will do all the work and you'll only load into memory the records you care about.

Reply:
:100:  thanks again!!